src/directives/automation/editor/dpEditorListsGrid.js

```diff
@@ -6,7 +6,6 @@
     .directive('dpEditorListsGrid', dpEditorListsGrid);
 
   dpEditorListsGrid.$inject = [
-    '$rootScope',
```

Comment:
I do not understand the change completely, but removing dependence on the `rootScope` and changing the event-based code to simple functions looks great!

Reply:
rootScope, is not used and it was added without needed

Reply:
Oh, in fact, it was used:

![image](https://github.com/FromDoppler/doppler-automation-editor/assets/1157864/046729de-8039-4d0f-abf4-5878f3f44dda)

And, for that reason, your change looks better yet.

Reply:
jaja, you right. I saw rootscope with light text after refactor